runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8513,7 +8513,7 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
 #if defined(J9VM_OPT_JITSERVER)
          if (that->_methodBeingCompiled->isOutOfProcessCompReq())
             {
-            OMRProcessorDesc JITClientProcessorDesc = TR::Compiler->target.cpu.getProcessorDescription();
+            OMRProcessorDesc JITClientProcessorDesc = that->_methodBeingCompiled->_compInfoPT->getClientData()->getOrCacheVMInfo(that->_methodBeingCompiled->_stream)->_processorDescription;
```

Comment:
`that` is a `CompilationInfoPerThreadBase` and you should be able to obtain the client data from there.